### PR TITLE
Email API

### PR DESCRIPTION
**Pull request by dciceman**
Created email api and wired up with the exist submission button click. After user completed their application and click on submit button. An email is send to both employer and certification team. 
Create email templates for employer and certification team.
Added unit test.

Create a new application a hit submit button for the email api test
#366  Email API

### CODE REVIEW

**Comment by dciceman** [comment]
@mgwalker The code check failed on the auto generated helppage.css. Please let me know if I need to do anything with that.

**Comment by mgwalker** [comment]
@dciceman In #424, I disabled csslint, but it hasn't been merged yet (still waiting for it to be reviewed).  For now, I went through the issues on CodeClimate and manually marked them as `invalid`.

File: DOL.WHD.Section14c.EmailApi/Business/Services/EmailService.cs [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+            }
+            finally
+            {
+                if(_smtpClient != null)
+                    _smtpClient.Dispose();
+            }
+
+            return success;
+        }        
+    }
```
Comment (mgwalker) [inline DOL.WHD.Section14c.EmailApi/Business/Services/EmailService.cs]: This class needs tests

Comment (dciceman) [reply DOL.WHD.Section14c.EmailApi/Business/Services/EmailService.cs]: The test class is here DOL.WHD.Section14c.EmailApi.Business.Tests.

File: DOL.WHD.Section14c.Business/Services/EmailService.cs [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+            var certificationEmailContents = ParseEmailContent(htmlString, Constants.CertificationTeamSection);
+            var employerEmailContents = ParseEmailContent(htmlString,Constants.EmployerSection);
+            var data = new Dictionary<string, EmailContent>();
+            // Prepare email contants based on the EmailReceiver value 
+            if (receiver == EmailReceiver.CertificationTeam || receiver == EmailReceiver.Both)
+                data.Add(Constants.CertificationEmailKey, certificationEmailContents);
+            if (receiver == EmailReceiver.Employer || receiver == EmailReceiver.Both)
+                data.Add(Constants.EmployerEmailKey, employerEmailContents);
+           
+            return data;
```
[the remark this answers is not in the document]
Comment (dciceman) [reply DOL.WHD.Section14c.Business/Services/EmailService.cs]: @mgwalker My only concern with this change is that the email subject and contents are in two separate locations. I will make the change as you suggested.

File: DOL.WHD.Section14c.Api/App_Data/EmailTemplate.html [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+Sincerely,<br /><br />
+
+Nicole Howard<br />    
+Supervisory Wage and Hour Investigator<br/>      
+FLSA Certification Team     
+            </email-body>
+        </employer-email-template>
+    </email-templates>
+</body>
+</html>
```
[the remark this answers is not in the document]
Comment (dciceman) [reply DOL.WHD.Section14c.Api/App_Data/EmailTemplate.html]: @mgwalker Handlebars is my new favorite tool. Thank you for suggesting it.

File: DOL.WHD.Section14c.EmailApi/Business/Services/EmailService.cs (hunk 1 above)
Comment (mgwalker) [reply DOL.WHD.Section14c.EmailApi/Business/Services/EmailService.cs]: You're right, my apologies!  There are two `EmailService` classes and two `EmailServiceTests` classes.  Having the same names and having all of the tests in the Test project makes it confusing what goes with what, but you're right!

File: DOL.WHD.Section14c.Test/Controllers/EmailControllerTests.cs [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+            var _emailServiceMock = new Mock<IEmailService>();
+            _emailController = new EmailController(_emailServiceMock.Object);
+        }
+
+        [TestMethod()]
+        public void SendEmailTest()
+        {
+            IHttpActionResult result = _emailController.SendEmail(_emailContent);
+            var contentResult = result as OkNegotiatedContentResult<Boolean>;
+            Assert.IsFalse(contentResult.Content);
```
Comment (mgwalker) [inline DOL.WHD.Section14c.Test/Controllers/EmailControllerTests.cs]: The goal of the test is to make sure the method under test does everything we expect it to do.  For `EmailController.SendEmail`, that means we need to be sure that it calls `IEmailService.SendEmail` and then uses the returned value.  To do that, we should setup the `IEmailService` mock to return a known value, not just the default value, and then assert that we got back the known value and that the mock's `SendEmail` function was called.  Something like this:

```C#
// Setup
_emailServiceMock.Setup(mock => mock.SendEmail(It.IsAny<EmailContent>()).Returns(true);

// Execute
IHttpActionResult result = _emailController.SendEmail(_emailContent);
var contentResult = result as OkNegotiatedContentResult<Boolean>;

// Assert
Assert.IsTrue(contentResult.Content);
_emailServiceMock.Verify(mock => mock.SendEmail(It.IsAny<EmailContent>()), Times.Once());
```

Comment (mgwalker) [reply DOL.WHD.Section14c.Test/Controllers/EmailControllerTests.cs]: ☝️  Once that's addressed, this will be good to go!

**Comment by mgwalker** [comment]
I just wanted to point out the 100% coverage on the EmailApi project.  :tada:

screen shot 2017-11-17 at 2 12 40 pm

**Pull request merged**